### PR TITLE
mongoose: unbreak on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/science/math/mongoose/default.nix [hunk 1]
```diff
@@ -16,17 +19,27 @@ stdenv.mkDerivation rec {
     sha256 = "0ymwd4n8p8s0ndh1vcbmjcsm0x2cc2b7v3baww5y6as12873bcrh";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/ScottKolo/Mongoose/commit/39f4a0059ff7bad5bffa84369f31839214ac7877.patch";
```
Comment (markuskowa) [inline pkgs/development/libraries/science/math/mongoose/default.nix:24]: Adding a descriptive `name` would be helpful (e.g. `name = "disable-coverage-option"`)
Can it be removed with the next release (maybe add a comment here)?

File: pkgs/development/libraries/science/math/mongoose/default.nix [hunk 2]
```diff
@@ -16,17 +19,27 @@ stdenv.mkDerivation rec {
...
+      sha256 = "sha256-V8lCq22ixCCzLmKtW6bUL8cvJFZzdgYoA4BFs4xYd3c=";
+    })
+  ];
+
   nativeBuildInputs = [
     cmake
   ];
 
+  cmakeFlags = lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
```
Comment (markuskowa) [inline pkgs/development/libraries/science/math/mongoose/default.nix:33]: There is only one option, i.e. `lib.option` is sufficient:
```suggestion
  cmakeFlags = lib.optional (stdenv.isDarwin && stdenv.isAarch64)
```

File: pkgs/development/libraries/science/math/mongoose/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/development/libraries/science/math/mongoose/default.nix:25]: Done.

File: pkgs/development/libraries/science/math/mongoose/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/development/libraries/science/math/mongoose/default.nix]: Done. Thanks!

**Pull request merged**